🐛 Fails to build on react-native 0.76 on Android

### What's happening?

Hi, when building using react-native 0.76, react-native-vision-camera fails with the following error:

```
C/C++: VisionCamera: Frame Processors: OFF!
C/C++: CMake Error at CMakeLists.txt:26 (add_library):
C/C++:   Target "VisionCamera" links to target "ReactAndroid::reactnativejni" but
C/C++:   the target was not found.  Perhaps a find_package() call is missing for an
C/C++:   IMPORTED target, or an ALIAS target is missing?
```

Both on new and old arch.

### Reproduceable Code

```tsx
N/A.

Check my repro here: https://github.com/hsjoberg/rn76-vision-camera-bug

It's simply a react-community cli project with react-native-vision-camera@4.5.3 added.
```


### Relevant log output

```shell
See above.
```


### Camera Device

```json
.
```


### Device

Emulator

### VisionCamera Version

4.5.3

### Can you reproduce this issue in the VisionCamera Example app?

I didn't try (⚠️ your issue might get ignored & closed if you don't try this)

### Additional information

- [ ] I am using Expo
- [ ] I have enabled Frame Processors (react-native-worklets-core)
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

--- a/package/android/CMakeLists.txt
+++ b/package/android/CMakeLists.txt
@@ -60,9 +60,21 @@ target_link_libraries(
         ${LOG_LIB}                          # <-- Logcat logger
         android                             # <-- Android JNI core
         ReactAndroid::jsi                   # <-- RN: JSI
-        ReactAndroid::reactnative           # <-- RN: React Native JNI bindings
         fbjni::fbjni                        # <-- fbjni
 )
+
+# Link react-native (different prefab between RN 0.75 and RN 0.76)
+if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
+    target_link_libraries(
+        ${PACKAGE_NAME}
+        ReactAndroid::reactnative                 # <-- RN: Native Modules umbrella prefab
+    )
+else()
+    target_link_libraries(
+        ${PACKAGE_NAME}
+        ReactAndroid::reactnativejni              # <-- RN: JNI Utils (e.g. CallInvoker)
+    )
+endif()
 
 # Optionally also add Frame Processors here
 message("VisionCamera: Frame Processors: ${ENABLE_FRAME_PROCESSORS}!")

--- a/package/android/CMakeLists.txt
+++ b/package/android/CMakeLists.txt
@@ -60,7 +60,7 @@ target_link_libraries(
         ${LOG_LIB}                          # <-- Logcat logger
         android                             # <-- Android JNI core
         ReactAndroid::jsi                   # <-- RN: JSI
-        ReactAndroid::reactnativejni        # <-- RN: React Native JNI bindings
+        ReactAndroid::reactnative           # <-- RN: React Native JNI bindings
         fbjni::fbjni                        # <-- fbjni
 )
 

